### ISSUE TITLE
wjecli validation -- suboptimal results

### ISSUE THREAD

**Issue by lzsiga** [netmail-open/wjelement#86]
Hi, I've just tried schema-validation, the results were suboptimal:

```
echo 'validate json-schema-draft-04.json' | wjecli json-schema-draft-04.json
definitions: extra property 'schemaArray' found.
definitions: extra property 'positiveInteger' found.
definitions: extra property 'positiveIntegerDefault0' found.
definitions: extra property 'simpleTypes' found.
definitions: extra property 'stringArray' found.
definitions failed validation
properties: extra property 'id' found.
properties: extra property '$schema' found.
...
```
As a start, the schema should be valid against itself, but even if it isn't, the error message should at least contain a line-number...

**Comment by penduin** [comment]
WJE's schema support is very outdated (even v4 isn't fully supported, clearly :^) and as much as I'd like to, I can't promise I'll have the time to bring it up to date.  Patches welcome, of course!

Line numbers might be more difficult; WJReader itself doesn't keep track of lines at all, simply chomping through and ignoring all whitespace completely (as determined by isspace()).  WJReader would need a new set of line-aware interfaces (for which the existing functions could become thin wrappers).  Above that, it would be easey to add properties in WJElement for line and character from which an element was originally read.

...once again this sounds like fun and I hope I get some time to do it, but in the meantime feel free and encouraged to dive in and take a crack at it yourself!

**Comment by minego** [comment]
I have considered this, and really it shouldn't be hard to add a line counter in the wjreader and increment it if we see a \n...

Maybe I'll try to play with it soon if I find a chance.

Jan 23, 2020 12:08:43 PM Owen Swerkstrom <notifications@github.com>:

**Comment by penduin** [comment]
true, it's nothing difficult.  just have to make sure it gets exposed back up to wjelement in a sane way, then make a pass through wjeschema's error reporting.

**Comment by lzsiga** [comment]
Thank you for you replies, maybe I'll able to offer some suggestions later.
Yours: L. Zs.
